lib/board.rb

```diff
@@ -0,0 +1,153 @@
+
+class Board
+  attr_accessor :board
+  def initialize(size=10)
+    @size = size
+    @board = Array.new(@size) {Array.new(@size){Cell.new}}
+    place_mines(generate_mines)
+    set_all_cell_adjacent_mines
+
+  end
+
+  def generate_mines
+    (0..((@size*@size)-1)).to_a.sample(@size)
+  end
+
+  def place_mines mine_locations
+    mine_locations.each do |x|
+      tens = x / 10
+      ones = x % 10
+      @board[tens][ones].mine = true
+    end
+  end
+
+  # def initialize_mines
+  #   place_mines(generate_mines)
+  # end
+
+  def set_all_cell_adjacent_mines
+    @board.each_with_index do |row, row_index|
+      @board[row_index].each_with_index do |column, column_index|
+        set_cell_adjacent_mines(row_index,column_index)
+      end
+    end
+  end
+
+  def is_mine? (row, column)
+    @board[row][column].mine
+  end
+
+  def set_cell_adjacent_mines(row, column)
+    around_array = [-1, 0, 1]
+    around_array.each do |row_offset|
+      around_array.each do |column_offset|
+        if (is_in_bound?(row+row_offset,column+column_offset) && is_mine?(row+row_offset,column+column_offset))
+          next if row_offset == row && column_offset == column 
+          @board[row][column].adjacent_mines += 1
+        end
+      end
+    end
+  end
+
+  def is_in_bound? (row_index, column_index)
+    if (row_index < 0 || column_index < 0)
+      return false
+    elsif (row_index > @size-1 || column_index > @size-1)
+      return false
+    else
+      return true
+    end
+  end
```

Comment:
I think you can simply this and refactor it as:
`row_index >= 0 && row_index < @size && column_index >= 0 && column_index < @size`